src/sqlfluff/rules/L039.py

```diff
@@ -66,4 +66,32 @@ def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
                         )
                 prev_newline = False
                 prev_whitespace = None
+
+            if (
+                seg.is_type("object_reference")
+                or seg.is_type("table_reference")
+                or seg.is_type("schema_reference")
+                or seg.is_type("database_reference")
+                or seg.is_type("index_reference")
+                or seg.is_type("extension_reference")
+                or seg.is_type("column_reference")
+                or seg.is_type("sequence_reference")
+                or seg.is_type("trigger_reference")
+            ):
```

Comment:
```suggestion
            if seg.is_type("object_reference"):
```
all these segments inherit from `ObjectReferenceSegment`, so the other conditions are redundant